assets/js/googlesitekit/datastore/site/reset.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { initializeAction } from 'assets/js/googlesitekit/data/utils';
+
+// Actions
+const FETCH_RESET = 'FETCH_RESET';
+const RECEIVE_RESET = 'RECEIVE_RESET';
+const RECEIVE_RESET_FAILURE = 'RECEIVE_RESET_FAILURE';
+
+export const INITIAL_STATE = {
+	isDoingReset: false,
+};
+
+export const actions = {
+	fetchReset() {
+		return {
+			payload: {},
+			type: FETCH_RESET,
+		};
+	},
+
+	receiveReset() {
+		return {
+			payload: {},
+			type: RECEIVE_RESET,
+		};
+	},
+
+	receiveResetFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_RESET_FAILURE,
+		};
+	},
+
+	*reset() {
+		try {
+			yield actions.fetchReset();
+			yield actions.receiveReset();
+			return initializeAction();
+		} catch ( err ) {
+			return actions.receiveResetFailed( err );
+		}
+	},
+};
+
+export const controls = {
+	[ FETCH_RESET ]: () => {
+		return API.set( 'core', 'site', 'reset' );
+	},
+};
+
+export const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case FETCH_RESET: {
+			return {
+				...state,
+				isDoingReset: true,
+			};
+		}
+
+		case RECEIVE_RESET_FAILURE: {
+			const { error } = action.payload;
+			return {
+				...state,
+				resetError: error,
```

Comment:
Same here: Here you put the error into the store separately, which is better, but I'd still wonder why, especially since it cannot be retrieved from the outside.

Moreover though, I'd say this is unnecessary here too. We should throw the error immediately because it is definitely unexpected.

Reply:
As mentioned above: throwing immediately here leads to odd flows for stores and connected components, so I think it's best not to do that. We have to decide on our error handling going forward, but let's leave that for later. The error handling for now is essentially left alone, but since we aren't using this anywhere yet it's safe to omit, knowing we need to do it for production.